include/deal.II/hp/fe_collection.h

```diff
@@ -371,6 +372,32 @@ namespace hp
     find_dominating_fe_in_subset(const std::set<unsigned int> &fes,
                                  const unsigned int            codim = 0) const;
 
+    /**
+     * Set functions determining the hierarchy of finite elements, i.e. a
+     * function @p next that returns the index of the finite element following
+     * the given one, and a function @p prev returning the preceding one.
+     *
+     * @note Both passed and returned indices have to be valid within the index
+     * range of this collection, i.e. within [0, size()).
+     */
+    void
+    set_hierarchy(const std::function<unsigned int(const unsigned int)> &next,
+                  const std::function<unsigned int(const unsigned int)> &prev);
+
+    /**
+     * Function returning the index of the finite element following the given
+     * @p fe_index in hierarchy.
```

Comment:
I think it would be useful to say what the *default* is and how one can change it.